### ISSUE TITLE
Panic on invalid code

### ISSUE THREAD

**Issue by marvinhagemeister** [swc-project/swc#9140]
### Describe the bug

SWC panics when it encounters this particular source that is invalid JS. The error can be reproduced in the SWC playground.

Originally reported at https://github.com/denoland/deno/issues/24417

### Input code

```typescript
https://fake.com

interface Fake {
}
```

### Playground link (or link to the minimal reproduction)

https://play.swc.rs/?version=1.6.7&code=H4sIAAAAAAAAA8soKSkottLXT0vMTtVLzs%2Fl4srMK0ktSktMTlVwA4opVHPVAgDVU9KdJAAAAA%3D%3D&config=H4sIAAAAAAAAA1WQPQ7CMAyFd05ReWZAHRi4AxMnsIJbBTU%2Fsh2JCvXuJOn%2FEiXfe8%2BO%2Fbs0DXzEwKP55Wt%2BRGQh3t6ZyOgVv5mAjpHEsI0K11VVKZJyokqmWQBF7klLiKS9te0SgCEEoYw7HIQW5qy33XhsaYKLTCJnY7Gi74ctf%2B7I6KUL7I6FmNDoAWRkXQysr5DYlELgo3vMtutu4uTVuqpj0uBQrYFFni7rWVuDC%2B9U%2F7RssGxpHvwOu2mdcZsHrDzXZF3f9Ae8XWptjQEAAA%3D%3D

### Expected behavior

SWC should not panic

### Version

1.6.8-nightly-20240704.3

**Comment by CPunisher** [comment]
From the AST of the code:
```json
"body": [
    {
      "type": "LabeledStatement",
      "span": {
        "start": 0,
        "end": 36,
        "ctxt": 0
      },
      "label": {
        "type": "Identifier",
        "span": {
          "start": 0,
          "end": 5,
          "ctxt": 0
        },
        "value": "https",
        "optional": false
      },
      "body": {
        "type": "TsInterfaceDeclaration",
        "span": {
          "start": 18,
          "end": 36,
          "ctxt": 0
        },
```

The `TsInterfaceDeclaration` within the `LabeledStatement` cannot be removed during `StripType`, thus it will be visited and trigger an unreachable panic ([Code](https://github.com/swc-project/swc/blob/181320a5554a1b29a64650bdc2c1097b67fe6e9a/crates/swc_ecma_transforms_typescript/src/strip_type.rs#L18)).

I attempted to change `unreachable_visit_mut_type!()` to `noop_visit_mut_type!()`, but it triggered the `standard_only_visit_mut!()` panic([Code](https://github.com/swc-project/swc/blob/181320a5554a1b29a64650bdc2c1097b67fe6e9a/crates/swc_ecma_compat_common/src/regexp.rs#L35)) of the RegExp Visitor(maybe other visitors).

Perhaps there are two possible solutions:
1. Patch `StripType` to remove every `TsInterfaceDeclaration`. This is how `tsc` do, and its compilation result is:
```ts
https: ;
```
2. Do not remove `TsInterfaceDeclaration` and find a way to resolve the subsequent panic. This is how `oxc` do, and its compilation result is:
```ts
https: interface Fake {}
```

**Comment by magic-akari** [comment]
fixed in #9364.
